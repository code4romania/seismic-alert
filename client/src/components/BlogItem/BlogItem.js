import { Col, Row, Typography } from 'antd';
import React from 'react';
import { Link, useRouteMatch } from 'react-router-dom';
import { Trans } from '@lingui/macro';

const { Paragraph, Text, Title } = Typography;

const BlogItem = ({ postDetails }) => {
  const { url } = useRouteMatch();
  const { image, title, published, slug } = postDetails;
  const previewText = postDetails.preview_text;
  const authorFullName = `${postDetails.author_first_name} ${postDetails.author_last_name}`;
  return (
    <Link to={`${url}/${slug}`} className="show-more">
      <Row className="blog-item">
        <Col className="blog-image" sm={24} md={12} lg={10}>
          <img src={image} alt={title} />
        </Col>
        <Col className="blog-meta" sm={24} md={12} lg={14}>
          <Title level={3} underline>
            {title}
          </Title>
          <Text type="secondary">
            <Trans>
              Published by {authorFullName} â€¢ {new Date(published).toLocaleDateString()}
            </Trans>
          </Text>
          <Paragraph>{previewText}</Paragraph>
        </Col>
      </Row>
    </Link>
  );
};

export default BlogItem;
