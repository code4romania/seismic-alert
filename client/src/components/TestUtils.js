import React from 'react';
import { i18n } from '@lingui/core';
import { I18nProvider } from '@lingui/react';
import { MemoryRouter } from 'react-router-dom';
import { messages as enMessages } from '../locales/en/messages';
import { messages as roMessages } from '../locales/ro/messages';

i18n.load({
  en: enMessages,
  ro: roMessages,
});

const LinguiWrapper = ({ children }) => <I18nProvider i18n={i18n}>{children}</I18nProvider>;

const RouterWrapper = ({ children }) => (
  <MemoryRouter>
    <LinguiWrapper>{children}</LinguiWrapper>
  </MemoryRouter>
);

export { LinguiWrapper, RouterWrapper };
