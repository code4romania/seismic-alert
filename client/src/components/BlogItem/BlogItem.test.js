import React from 'react';
import { i18n } from '@lingui/core';
import { act, render } from '@testing-library/react';
import { RouterWrapper } from '../TestUtils';
import BlogItem from './BlogItem';

describe('BlogItem component', () => {
  const postDetails = {
    title: 'Title',
    image: 'image-url',
    preview_text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit',
    published: new Date(0),
    author_first_name: 'First',
    author_last_name: 'Last',
  };
  const size = 8;

  it('should be translated correctly in English', async () => {
    act(() => {
      i18n.activate('en');
    });
    const container = render(<BlogItem size={size} postDetails={postDetails} />, {
      wrapper: RouterWrapper,
    });

    expect(container.baseElement).toMatchSnapshot();
  });

  it('should be translated correctly in Romanian', async () => {
    act(() => {
      i18n.activate('ro');
    });
    const container = render(<BlogItem size={size} postDetails={postDetails} />, {
      wrapper: RouterWrapper,
    });
    expect(container.baseElement).toMatchSnapshot();
  });
});
