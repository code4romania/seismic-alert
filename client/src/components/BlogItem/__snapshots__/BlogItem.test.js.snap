// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogItem component should be translated correctly in English 1`] = `
<body>
  <div>
    <a
      class="show-more"
      href="//undefined"
    >
      <div
        class="ant-row blog-item"
      >
        <div
          class="ant-col blog-image ant-col-sm-24 ant-col-md-12 ant-col-lg-10"
        >
          <img
            alt="Title"
            src="image-url"
          />
        </div>
        <div
          class="ant-col blog-meta ant-col-sm-24 ant-col-md-12 ant-col-lg-14"
        >
          <h3
            class="ant-typography"
          >
            <u>
              Title
            </u>
          </h3>
          <span
            class="ant-typography ant-typography-secondary"
          >
            Published by First Last • 1/1/1970
          </span>
          <div
            class="ant-typography"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit
          </div>
        </div>
      </div>
    </a>
  </div>
</body>
`;

exports[`BlogItem component should be translated correctly in Romanian 1`] = `
<body>
  <div>
    <a
      class="show-more"
      href="//undefined"
    >
      <div
        class="ant-row blog-item"
      >
        <div
          class="ant-col blog-image ant-col-sm-24 ant-col-md-12 ant-col-lg-10"
        >
          <img
            alt="Title"
            src="image-url"
          />
        </div>
        <div
          class="ant-col blog-meta ant-col-sm-24 ant-col-md-12 ant-col-lg-14"
        >
          <h3
            class="ant-typography"
          >
            <u>
              Title
            </u>
          </h3>
          <span
            class="ant-typography ant-typography-secondary"
          >
            Publicat de First Last • 1/1/1970
          </span>
          <div
            class="ant-typography"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit
          </div>
        </div>
      </div>
    </a>
  </div>
</body>
`;
